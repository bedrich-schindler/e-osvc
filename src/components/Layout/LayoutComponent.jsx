import PropTypes from 'prop-types';
import React from 'react';
import { Header } from '../Header';
import { Menu } from '../Menu';
import { NotificationCenter } from '../NotificationCenter';
import { Timer } from '../Timer';
import styles from './styles.scss';

class LayoutComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isMenuOpened: false,
    };

    this.handleOpenMenu = this.handleOpenMenu.bind(this);
    this.handleCloseMenu = this.handleCloseMenu.bind(this);
  }

  handleOpenMenu() {
    this.setState({ isMenuOpened: true });
  }

  handleCloseMenu() {
    this.setState({ isMenuOpened: false });
  }

  render() {
    const {
      children,
      logout,
      user,
    } = this.props;
    const { isMenuOpened } = this.state;

    return (
      <div>
        <div className={styles.root}>
          <Header
            isMenuOpened={isMenuOpened}
            onLogout={logout}
            onOpenMenu={this.handleOpenMenu}
            title="eOSVÄŒ"
            user={user}
          />
          <Menu
            isMenuOpened={isMenuOpened}
            onCloseMenu={this.handleCloseMenu}
          />
        </div>
        <main className={styles.content}>
          <NotificationCenter />
          {children}
          <Timer />
        </main>
      </div>
    );
  }
}

LayoutComponent.defaultProps = {
  children: null,
  floatingActions: null,
};

LayoutComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.element,
  ]),
  logout: PropTypes.func.isRequired,
  user: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default LayoutComponent;
