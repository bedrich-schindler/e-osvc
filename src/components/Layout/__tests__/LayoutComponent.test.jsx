import React from 'react';
import { shallow } from 'enzyme';
import LayoutComponent from '../LayoutComponent';

describe('rendering', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <LayoutComponent
        title="Title"
        user={{ name: 'Name' }}
        logout={() => {}}
      >
        <p>Content</p>
      </LayoutComponent>,
    );

    expect(tree).toMatchSnapshot();
  });
});
