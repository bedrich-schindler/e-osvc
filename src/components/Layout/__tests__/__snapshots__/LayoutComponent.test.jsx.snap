// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<div>
  <div
    className="root"
  >
    <WithStyles(withRouter(Header))
      isMenuOpened={false}
      onLogout={[Function]}
      onOpenMenu={[Function]}
      title="eOSVÄŒ"
      user={
        Object {
          "name": "",
        }
      }
    />
    <WithStyles(withRouter(Menu))
      isMenuOpened={false}
      onCloseMenu={[Function]}
    />
  </div>
  <main
    className="content"
  >
    <Connect(NotificationCenterComponent) />
    <Styled(MuiBox)
      mb={5}
      mt={2}
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        direction="row"
        justify="space-between"
        spacing={5}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <h1
            style={
              Object {
                "margin": 0,
              }
            }
          >
            Title
          </h1>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </Styled(MuiBox)>
    <p>
      Content
    </p>
    <Connect(TimerComponent) />
  </main>
</div>
`;
